### DISCUSSION TITLE
What is difference between `"@repo/math": "*"` and `"@repo/math": "workspace:^"`?

### DISCUSSION THREAD

**Discussion by cjayyy** [vercel/turborepo#9523] [Help] [answered]
### Summary

When adding internal package to another package in Yarn, [Turbo documentation recommends](https://turbo.build/repo/docs/crafting-your-repository/creating-an-internal-package#add-the-package-to-an-application) to add it as wildcard version (`"@repo/math": "*"`). [Yarn docs](https://yarnpkg.com/features/workspaces#cross-references) on the other hand recommend to use `workspace:` protocol (`"@repo/math": "workspace:^"`) which seems safer and more readable. 

Is there any difference in these two ways for Turborepo? 
Why doesn't Turbo recommend the Yarn recommended way in its documentation?

**Comment by anthonyshew** [comment, marked as answer]
They both do the same thing, but are for different Yarn versions. We show the `*` version in the docs because that's how Yarn 1 works, and way more Turborepo users use Yarn 1 compared to Yarn 2+.

For Yarn 1, you could also pin a version and update that as needed, but you'll notice it also doesn't have any `workspace` moniker in the docs: https://classic.yarnpkg.com/en/docs/workspaces
